test_invader_payment/tests/test_stripe.py

```diff
@@ -65,9 +72,31 @@ def test_confirm_payment_two_step(self):
         # on server side so the simpliest solution is to
         # hack the next response
         self._alter_next_response({"status": "succeeded"})
-        result = self.service.confirm_payment(
-            "demo_partner",
-            payment_mode=self.stripe_method.id,
-            stripe_payment_intent_id=stripe_payment_intent_id,
+        result = self.service.dispatch(
+            "confirm_payment",
+            params={
+                "target": "demo_partner",
+                "payment_mode_id": self.payment_mode.id,
+                "stripe_payment_intent_id": stripe_payment_intent_id,
+            },
         )
         self.assertEqual(result, {"success": True})
+
+    def test_wrong_provider_confirm(self):
+        self.payment_mode = self.env.ref(
```

Comment:
@sebastienbeau `payment_mode`should be scoped to the method not to the instance...
`s/self.payment_mode/payment_mode_check/`